code/scripting/api/objs/subsystem.cpp

```diff
@@ -587,6 +588,60 @@ ADE_VIRTVAR(TurretLocked, l_Subsystem, "boolean", "Whether the turret is locked.
 	return ade_set_args(L, "b", (sso->ss->weapons.flags[Ship::Weapon_Flags::Turret_Lock]));
 }
 
+ADE_VIRTVAR(TurretLockedWithTimestamp, l_Subsystem, "boolean", "Behaves like TurretLocked, but when the turret is freed, the next-fire timestamp is set to be consistent with SEXP behavior.", "boolean", "True if turret is locked, false otherwise")
```

Comment:
I would be more specific and say that this specifically will add a short random delay before firing (as as to be consistent yadda yadda), a user isn't going to know what "the next-fire timestamp is set" implies. Ditto below.

Although... the next fire time stamp is settable via script *anyway*, so I have to wonder, if the scripter really wants this short delay perhaps it would be best to simply let them do it? Rather than complicate the API with this tiny convenience.

Reply:
I think it's worthwhile to have a script operation that behaves exactly like the sexp, even if it's equivalent to two existing script operations.  Plus, if this operation is used a lot, the scripter would otherwise have to keep the extra operation in mind or use `mn.evaluateSEXP`.

I'll update the description; that's a good idea.